cypress/e2e/settings.cy.js

```diff
@@ -1,32 +1,48 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import { SettingsPageObject } from '../support/pages/settings.pageObject';
+
+const settingsPage = new SettingsPageObject();
+
 describe('Settings page', () => {
+  let user;
   before(() => {
-
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
   });
 
   beforeEach(() => {
-
+    cy.visit('/');
+    cy.login(user.email, user.username, user.password);
+    settingsPage.visit();
   });
 
   it('should provide an ability to update username', () => {
-
+    settingsPage.typeUsername(user.username);
+    settingsPage.submitForm();
+    settingsPage.assertProfilePage(user.username);
```

Comment:
Ensure that `assertProfilePage` correctly verifies that the username has been updated. This might involve checking the displayed username on the profile page to confirm the update.